.github/workflows/ig-analysis.yml

```diff
@@ -0,0 +1,34 @@
+name: Analyze Go Package Sizes and Build
+
+on:
+  pull_request:
+jobs:
+  analyze-and-build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Set up Go
+        uses: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0
+        with:
+          go-version: '1.23.4'
+          cache: true
+        id: go
+
+      - name: Install dependencies
+        run: |
+          go mod tidy
+
+      - name: Install go-size-analyzer
+        run: |
+          go get github.com/Zxilly/go-size-analyzer/cmd/gsa@latest
```

Comment:
Since this has a recent release [v1.7.6](https://github.com/Zxilly/go-size-analyzer/releases/tag/v1.7.6), should we use it instead of the latest tag?